### PR TITLE
Shots and feature map exposure

### PR DESCRIPTION
**Pull request by gcattan**
Expose the number of shots and the feature map (part of issue #13)

### CODE REVIEW

**Review by sylvchev — APPROVED** [review]
Nice work!

**Review by qbarthelemy — CHANGES_REQUESTED** [review]
LGTM! Only minor comments.

File: pyriemann_qiskit/utils/hyper_params_factory.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+def gen_zz_feature_map(reps=2, entanglement='linear'):
+    """Return a callable that generate a ZZFeatureMap.
+    A feature map encodes data into a quantum state.
+    A ZZFeatureMap is a second-order Pauli-Z evolution circuit.
+
+    Parameters
+    ----------
+    reps : int
+        The number of repeated circuits, has a min. value of 1.
+    entanglement : Union[str, List[List[int]], Callable[[int], List[int]]]
```
[the remark this answers is not in the document]
Comment (gcattan) [reply pyriemann_qiskit/utils/hyper_params_factory.py:14]: This is a union between a string and list of indices. I have now improved the description.

File: pyriemann_qiskit/utils/hyper_params_factory.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+def gen_zz_feature_map(reps=2, entanglement='linear'):
+    """Return a callable that generate a ZZFeatureMap.
+    A feature map encodes data into a quantum state.
+    A ZZFeatureMap is a second-order Pauli-Z evolution circuit.
+
+    Parameters
+    ----------
+    reps : int
+        The number of repeated circuits, has a min. value of 1.
```
[the remark this answers is not in the document]
Comment (gcattan) [reply pyriemann_qiskit/utils/hyper_params_factory.py]: I wanted to say that the value has to be greater or equal to 1. 
I have now updated the description to put it more clear.

**Comment by gcattan** [comment]
Thank you for the comments. There is still some problem with the pipeline. I should investigate it deeper, but it seems that `git` failed to be installed on the doc pipeline. If you have an idea do not hesitate :)

**Review by qbarthelemy — CHANGES_REQUESTED** [review]
Again, some minor remarks.

More generally:

1 - don't merge master in your branch, but **rebase** your branch on master; otherwise, it is really difficult to understand if modifications com from branch and from master.

2 - to easily update your fork master on the pyRiemann-qiskit master, do:
`git remote add upstream https://github.com/pyriemann/pyriemann-qiskit.git`
Then, before creating a branch, pull from `upstream`.

See SciPy guidelines: https://docs.scipy.org/doc/scipy/reference/dev/contributor/development_workflow.html

**Comment by qbarthelemy** [comment]
> There is still some problem with the pipeline. I should investigate it deeper, but it seems that git failed to be installed on the doc pipeline. If you have an idea do not hesitate :)

I think that the problem comes from module `datasets` of pyRiemann, not declared in init file. Its import fails. This should be corrected after merge of
https://github.com/pyRiemann/pyRiemann/pull/149/commits/91c62e1c6d2c7ad789c186c65ca39a769d33016f
@sylvchev

File: pyriemann_qiskit/utils/hyper_params_factory.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+from qiskit.circuit.library import ZZFeatureMap
+
+
+def gen_zz_feature_map(reps=2, entanglement='linear'):
```
[the remark this answers is not in the document]
Comment (gcattan) [reply pyriemann_qiskit/utils/hyper_params_factory.py:4]: Thanks! I added a couple of tests and removed the `pairwise` option which was not available in this version.

File: tests/test_utils_hyper_params_factory.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+def test_gen_zz_feature_map_entangl_invalid_value():
+    """Test gen_zz_feature_map with uncorrect value"""
+    feature_dim = 2
+    feature_map = gen_zz_feature_map(entanglement="invalid")(feature_dim)
+    try:
+        feature_map.parameters
+        raise Exception("Invalid option should raise a ValueError exception")
+    except ValueError:
+        pass
```
Comment (qbarthelemy) [inline tests/test_utils_hyper_params_factory.py:41]: To test if a function raises an error, you can use:
```
with pytest.raises(ValueError):
    my_function.do_a_stuff_raising_an_error()
```

File: tests/conftest.py [hunk 5]
```diff
@@ -65,3 +65,35 @@ def get_feats(rndstate):
     def _gen_feat(n_samples, n_features):
         return generate_feat(n_samples, n_features, rndstate)
     return _gen_feat
+
+
+def _get_linear_entanglement(n_qbits_in_block, feature_dim):
```
Comment (qbarthelemy) [inline tests/conftest.py:70]: For consistency in notations, you should use `n_features` instead of `feature_dim`.

**Review by qbarthelemy — CHANGES_REQUESTED** [review]
Great job!
Last comments, then it will be ok for me.

**Comment by gcattan** [comment]
Thank you for the useful comments! There is still some CI/CD error. I will wait for the aforementioned commit to be merged and come back to this if required :)

**Review by qbarthelemy — APPROVED** [review]
Thx @gcattan !

**Pull request merged**